server/start.go

```diff
@@ -290,26 +288,23 @@ func startInProcess(ctx *Context, clientCtx client.Context, appCreator types.App
 		}
 	}
 
-	TrapSignal(func() {
-		if tmNode.IsRunning() {
-			_ = tmNode.Stop()
-		}
+	WaitForQuitSignals()
 
-		if cpuProfileCleanup != nil {
-			cpuProfileCleanup()
-		}
+	if tmNode.IsRunning() {
+		_ = tmNode.Stop()
+	}
 
-		if apiSrv != nil {
-			_ = apiSrv.Close()
-		}
+	if cpuProfileCleanup != nil {
+		cpuProfileCleanup()
+	}
 
-		if grpcSrv != nil {
-			grpcSrv.Stop()
-		}
+	if apiSrv != nil {
+		_ = apiSrv.Close()
+	}
 
-		ctx.Logger.Info("exiting...")
-	})
+	if grpcSrv != nil {
+		grpcSrv.Stop()
+	}
```

Comment:
Ok can we at least group these in a single cleanup function?

Reply:
made it into a defer expression now.